examples/column/zalesak_fct_advection.jl

```diff
@@ -0,0 +1,144 @@
+using Test
+using LinearAlgebra
+using OrdinaryDiffEq: ODEProblem, solve
+using DiffEqBase
+using ClimaTimeSteppers
+
+import ClimaCore:
+    Fields,
+    Domains,
+    Topologies,
+    Meshes,
+    DataLayouts,
+    Operators,
+    Geometry,
+    Spaces
+
+
+# Advection Equation, with constant advective velocity (so advection form = flux form)
+# ∂_t y + w ∂_z y  = 0
+# the solution translates to the right at speed w,
+# so at time t, the solution is y(z - w * t)
+
+# visualization artifacts
+ENV["GKSwstype"] = "nul"
+using ClimaCorePlots, Plots
+Plots.GRBackend()
+dir = "zalesak_fct_advection"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+function linkfig(figpath, alt = "")
+    # buildkite-agent upload figpath
+    # link figure in logs if we are running on CI
+    if get(ENV, "BUILDKITE", "") == "true"
+        artifact_url = "artifact://$figpath"
+        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+    end
+end
+
+function f!(dydt, y, parameters, t, alpha, beta)
+
+    (; w, A, y_td) = parameters
+    y = y.y
+    dydt = dydt.y
+
+    first_order_fluxᶠ = Operators.UpwindBiasedProductC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    third_order_fluxᶠ = Operators.Upwind3rdOrderBiasedProductC2F(
+        bottom = Operators.ThirdOrderOneSided(),
+        top = Operators.ThirdOrderOneSided(),
+    )
+    divf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0.0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0.0))),
+    )
+    FCTZalesak = Operators.FCTZalesak(
+        bottom = Operators.ThirdOrderOneSided(),
+        top = Operators.ThirdOrderOneSided(),
+    )
+
+    @. y_td = beta * dydt - alpha * divf2c(first_order_fluxᶠ(w, y))
+    @. dydt =
+        y_td -
+        alpha * divf2c(
+            FCTZalesak(
+                third_order_fluxᶠ(w, y) - first_order_fluxᶠ(w, y),
+                y / alpha,
+                y_td / alpha,
+            ),
+        )
+
+    return dydt
+end
+
+FT = Float64
+t₀ = FT(0.0)
+t₁ = FT(1.0)
+z₀ = FT(0.0)
+zₕ = FT(1.0)
+z₁ = FT(1.0)
+speed = FT(1.0)
+
+n = 2 .^ 6
+
+domain = Domains.IntervalDomain(
+    Geometry.ZPoint{FT}(-π),
+    Geometry.ZPoint{FT}(π);
+    boundary_names = (:bottom, :top),
+    # periodic = true,
+)
+mesh = Meshes.IntervalMesh(domain, nelems = n)
+cs = Spaces.CenterFiniteDifferenceSpace(mesh)
+fs = Spaces.FaceFiniteDifferenceSpace(cs)
+zc = Fields.coordinate_field(cs)
+
+# Unitary, constant advective velocity
+w = Geometry.WVector.(speed .* ones(FT, fs))
+
+# Define a pulse wave or square wave
+pulse(z, t, z₀, zₕ, z₁) = abs(z.z - speed * t) ≤ zₕ ? z₁ : z₀
```

Comment:
I would generalize this (and the comment above), for the case in which you want to run this for one (or more) periods

Reply:
I think the current setup is fine, I copy the test as your bb_fct test. I only run 1 period to understand the algorithm